src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageProcessor.java

```diff
@@ -84,27 +82,18 @@ private MessageProcessingResult processCreateCommand(IMessage message) {
 
         try {
             payment = paymentMessageParser.parse(message.getMessageBody());
-            paymentMessageHandler.handlePaymentMessage(payment);
+            paymentMessageHandler.handlePaymentMessage(payment, message.getMessageId());
+
             log.info(
                 "Processed payment message with ID {}. Envelope ID: {}",
                 message.getMessageId(),
                 payment.envelopeId
             );
+
             return new MessageProcessingResult(SUCCESS);
         } catch (InvalidMessageException ex) {
             log.error("Rejected payment message with ID {}, because it's invalid", message.getMessageId(), ex);
             return new MessageProcessingResult(UNRECOVERABLE_FAILURE, ex);
-        } catch (PayHubClientException ex) {
-            if (ex.getStatus() == HttpStatus.CONFLICT) {
```

Comment:
Handling conflict at this level is too late - we need to decide whether to call CCD or not in `paymentMessageHandler`.